src/client.rs

```diff
@@ -185,12 +188,117 @@ impl Stream for Events {
     }
 }
 
+pub struct Client {
+    pub framework_id: String,
+    stream_id: String,
+    pub events: Events,
+}
+
+impl fmt::Debug for Client {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Client {{ framework_id: {}, stream_id: {} }}", self.framework_id, self.stream_id)
+    }
+}
+
+header! { (MesosStreamIdHeader, "Mesos-Stream-Id") => [String] }
+
+impl Client {
+    pub fn connect(
+        handle: &Handle,
+        uri: hyper::Uri,
+        framework_info: mesos::FrameworkInfo,
+    ) -> Box<Future<Item = Self, Error = failure::Error>> {
+        // Mesos subscribe call
+        let call = Self::subscribe(framework_info);
+        let request = Self::request_for(uri, call);
+
+        // Call Mesos
+        let http_client = hyper::Client::new(&handle);
+        let s = http_client
+            .request(request)
+            .map_err(failure::Error::from)
+            .and_then(move |res: hyper::Response| {
+                println!("Response status: {}", res.status());
```

Comment:
https://docs.rs/log/0.4.1/log/ especially since this is a library

Reply:
Done.